pkg/jx/cmd/step/create/step_create_meta_pipeline.go

```diff
@@ -0,0 +1,318 @@
+	cmd.Flags().StringVarP(&options.Trigger, "trigger", "t", string(pipelineapi.PipelineTriggerTypeManual), "The kind of pipeline trigger")
+	cmd.Flags().StringVarP(&options.ServiceAccount, "service-account", "", "tekton-bot", "The Kubernetes ServiceAccount to use to run the pipeline")
+
+	cmd.Flags().StringVarP(&options.OutDir, "output", "o", "out", "The directory to write the output to as YAML")
+	cmd.Flags().BoolVarP(&options.NoApply, "no-apply", "", false, "Disables creating the pipeline resources in the kubernetes cluster and just outputs the generated Task to the console or output file")
+	cmd.Flags().BoolVarP(&options.ViewSteps, "view", "", false, "Just view the steps that would be created")
+
+	options.AddCommonFlags(cmd)
+	return cmd
+}
+
+// Run implements this command
+func (o *StepCreatePipelineOptions) Run() error {
+	err := o.validateCommandLineFlags()
+	if err != nil {
+		return err
+	}
+
+	gitInfo, _ := gits.ParseGitURL(o.GitCloneURL)
+
+	tektonClient, jxClient, kubeClient, ns, err := o.getClientsAndNamespace()
+	if err != nil {
+		return err
+	}
+
+	podTemplates, err := o.getPodTemplates(kubeClient, ns, apps.AppPodTemplateName)
+	if err != nil {
+		return errors.Wrap(err, "unable to retrieve pod templates")
+	}
+
+	pipelineResourceName := tekton.PipelineResourceName(gitInfo, o.Branch, o.Context)
+	buildNumber, err := tekton.GenerateNextBuildNumber(tektonClient, jxClient, ns, gitInfo, o.Branch, retryDuration, pipelineResourceName)
+	if err != nil {
+		return errors.Wrap(err, "unable to determine next build number")
+	}
+
+	if o.Verbose {
+		log.Infof("creating meta pipeline CRDs for build %s of repository '%s/%s'", buildNumber, gitInfo.Organisation, gitInfo.Name)
+	}
+	extendingApps, err := metapipeline.GetExtendingApps(jxClient, ns)
+	if err != nil {
+		return err
+	}
+
+	pipelineName := tekton.PipelineResourceName(gitInfo, o.Branch, o.Context)
+	labels := o.buildLabels(gitInfo)
+	envVars := o.buildEnvVars()
+	crdCreationParams := metapipeline.CRDCreationParameters{
+		Namespace:      ns,
+		Context:        o.Context,
+		PipelineName:   pipelineName,
+		PipelineKind:   o.PipelineKind,
+		BuildNumber:    buildNumber,
+		GitInfo:        gitInfo,
+		Branch:         o.Branch,
+		PullRef:        o.PullRefs,
+		Revision:       o.Revision,
+		SourceDir:      o.SourceName,
+		PodTemplates:   podTemplates,
+		Trigger:        o.Trigger,
+		ServiceAccount: o.ServiceAccount,
+		Labels:         labels,
+		EnvVars:        envVars,
+		Apps:           extendingApps,
+	}
+	tektonCRDs, err := metapipeline.CreateMetaPipelineCRDs(crdCreationParams)
+	if err != nil {
+		return errors.Wrap(err, "failed to generate Tekton CRDs for meta pipeline")
+	}
+
+	err = o.handleResult(tektonClient, jxClient, tektonCRDs, buildNumber, o.Branch, o.PullRefs, ns, gitInfo)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *StepCreatePipelineOptions) validateCommandLineFlags() error {
```

Comment:
@abayer, I'd really like to end up at a stage where it is clearly defined what parameters are needed. Also,. easy validation should occur early on.

Reply:
Agreed.